Configure OkCandidate Platform for Cucumber

rydente opened the pull request:
Introduce Cucumber as an acceptance testing and feature collaboration
system to OkCandidate so we can define and verify the system from a
functional perspective based on the users' perspectives.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

Commit:
Configure OkCandidate Platform for Cucumber

Introduce Cucumber as an acceptance testing and feature collaboration
system to OkCandidate so we can define and verify the system from a
functional perspective based on the users' perspectives.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

wbprice commented:
What are the specific challenges with using the JavaScript implementation of Cucumber?

rydente commented:
@wbprice From reading the recent documentation and checking up on collaborating toolsets, the teams involved may have mitigated them, but last I used it (November of last year):

* every step had to return a Promise;
* the headless browser (Zombie.js) hadn't had an update in a long time;
* Zombie didn't work properly in certain cases (cookies and ignoring self-signed certificates came to mind);

As a result, it read poorly (generally, features, scenarios and steps should read very, very clearly) and had lots of little bugs which made implementing steps take far longer than normal, made them flakier and harder to debug when they went wrong.

However, I'm still in the process of writing up a bdd-js PR that I will send out soon, at latest tomorrow, so would you want to wait for that to come in and make a decision then?

**EDIT**: It doesn't seem as bad as it used to.  I'll put up two gists to compare how tests look and link to each.

[Here](https://gist.github.com/rydente/a71b9a41cee592a2ea98fdb9b5994ae9)'s the Ruby gist.  The related PR below gives the Cucumber.js example.

Commit:
Incorporate latest changes

Merge branch 'develop' into bdd-ruby to integrate the Cucumber setup
with the latest implementation.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

Commit:
Incorporate latest changes

Merge branch 'develop' into bdd-ruby to integrate the Cucumber setup
with the latest implementation.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

wbprice commented on Gemfile:
```diff
@@ -0,0 +1,27 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
```
We don't need this here if we have a LICENSE file in the project root.

wbprice commented:
Closing this because we're probably going to use the JS implementation of Cucumber.
